### DISCUSSION TITLE
New page title implementation

### DISCUSSION THREAD

**Discussion by zoulamZero** [toeverything/blocksuite#603] [General]
I tested and found that the current title implemented using textarea has several problems.
Cursor not visible in collaborative mode, misbehavior when there is an empty line in the first line (I demonstrated it to [Flrande](https://github.com/Flrande)).

The following is my understanding of the definition of blocksuite title (please correct me if I am wrong):
The title corresponds to the namespace of the page, and the title is not displayed after entering edgeless mode. Multiple Frames can be added under edgeless mode, which will be merged into one document after edgeless mode is turned off (the definition in the block suite is page), title is the overall name of Frames;

Is it possible to try to define title using quill-block , only need to satisfy the following behavior:
1. The title should be unique in the page and always exist (the content of the title can be deleted, but the block cannot be deleted, that is, there is at least a placeholder for the title);
2. Ensure title does not exist in edgeless mode;
3. Remove the drag behavior of quill-block(title);

<details>
<summary>中文</summary>
<br/>
我测试发现当前使用 textarea 实现的 title 存在几个问题:

在协作模式下无法看到光标，首行存在空行的时候出现行为异常（我向 [Flrande](https://github.com/Flrande)  演示过）。

下面是个人对 blocksuite title 的定义的理解（如有错误请指正）：
title 对应的是 page 的命名空间，在进入 edgeless mode 后是不显示 title 的。edgeless mode 下面可以添加多个 Frame，在关闭 edgeless mode 之后会被合并为一个文档（在block suite里面的定义是 page），title 是 Frames 的整体命名；

是否可以尝试使用 quill-block 定义 title ，只需要满足以下行为：
1. title 在 page 中应该是独一无二的，且始终存在（title 的内容可以被删除，但是 block 无法被删除，即至少存在 title 的 placeholder）；
2. 保证在 edgeless mode 下 title 不存在；
3. 移除 quill-block(title) 的 drag 行为；
</details>

**Comment by doodlewind** [comment]
> The title corresponds to the namespace of the page, and the title is not displayed after entering edgeless mode. Multiple Frames can be added under the edgeless mode, which will be merged into one document after the edgeless mode is turned off (the definition in the block suite is page), title is the overall name of Frames.

Yes, that's the basic assumption of the page title!

As for a new implementation based on Quill, this could be non-trivial since we also need to handle the hotkey binding and clipboard behavior (to disable rich-format pasted-in). Also, the data type of the current title is a string, but our Quill component uses `Y.Text`. So there might be a breaking change.

If we are going to make this happen, there could be another option that may be worth the effort and bring a much higher impact: Compared to Quill, the basic arch in BlockSuite has enabled us to use lighter "rich text components". So for a Quill alternative rich-text component, making it work in the title makes more sense to me - after all, this component can be nothing more than a minimal collaborative `<textarea>` that supports delta format, which is also possible to be reused in our future comment box. I have talked with @Flrande and hope he can give it a try recently.
